src/main/java/net/sf/mzmine/modules/peaklistmethods/io/gnpsexport/GNPSExportTask.java

```diff
@@ -159,7 +176,7 @@ private void export(PeakList peakList, FileWriter writer, File curFile)
                 
                 String mass = Double.toString(row.getAverageMZ());
                 if (mass != null) writer.write("PEPMASS=" + mass + newLine);                                                                      
-                writer.write("CHARGE=1"+newLine);
+                writer.write("CHARGE="+row.getRowCharge()+"+"+newLine);
```

Comment:
I guess you can use something like
````
row.getBestPeak().getDataFile().getScan( row.getBestPeak().getRepresentativeScanNumber()).getPolarity().asSingleChar() 
````
But don't forget to check for null values and Polarity.UNKNOWN etc.